### PR TITLE
Assignment5: Austin & Brady

### PR DESCRIPTION
**Pull request by BDullanty**
@bdullanty @austinh5544

### CODE REVIEW

File: CircluarLinkedList/Node.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        newNode.Next = lastNode;
+        Next = newNode;
+    }
+
+    public override string? ToString()
+    {
+        return Value?.ToString();
+    }
+
+    public bool Exists(T? expectedValue)
```
Comment (LeoMarlow) [inline CircluarLinkedList/Node.cs:32]: This is super clean

File: CircluarLinkedList/Node.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        Node<T> current = this;
+
+        do
+        {
+            if ((current.Value is null && expectedValue is null) || current.Value?.Equals(expectedValue) == true)
```
Comment (LeoMarlow) [inline CircluarLinkedList/Node.cs:38]: (current.Value == expectedValue) fix 

I dont believe that you need a double check here.

**Review by LeoMarlow — COMMENTED** [review]
Create a class library project called "GenericsHomework.". ✔
Create a node class that can contain a value of any type and points to the next node and traversing the next node points back to the first item.✔
Define the Node class✔
The values should be homogeneous.✔
Include a constructor that takes a value. (No validation is necessary on the value). ✔
Add a ToString() override that writes out the value's ToString() result. ✔
Add a Next property that references the next node or else refers back to itself if there are no other nodes in the list. ✔
The Next property should be non-nullable (careful to follow the non-nullable property guidelines) ✔
The Next property setter should be private. ✔
Add an Append method that takes a value and appends a new Node instance after the current node (by invoking the Next property). ✔
Add a Clear method that effectively removes all items from a list except the current node. Pay attention as to whether you should be concerned with the following:
Whether it is sufficient to only set Next to itself ✔
Whether to set the removed items to circle back on themselves. In other words, whether to close the loop of the removed items. (Provide a test to show why this is required if it is required). ✔
Given there is a circular list of items, provide a comment to indicate whether you need to worry about garbage collection because all the items point to each other and, therefore, may never be garbage collected. ✔
Create an Exists method to test whether a value exists in the list. ✔
Throw an error on an attempt to Append a duplicate value. (Make sure you test for this case) ✔
You should not rely on any BCL generic classes for your implementation. ✔
Extra Credit
Do one of the following two options (or both if you want extra, extra credit) :)

Implement a VennDiagram structure that contains n Circles that only contains homogenous reference types of any type. ❌
Each circle contains n items and each item can belong to one or more Circle instances.
You are not required to use a Node from earlier in the homework for your Venn diagram implementation.
You are welcome to use existing BCL generic classes for the extra credit.
Implement System.Collections.Generic.ICollection<T> on the Node class ❌

Good job guys only one small issue with a logic but great job !

File: CircluarLinkedList/Node.cs (hunk 1 above)
Comment (BDullanty) [reply CircluarLinkedList/Node.cs:32]: Thank you :)

File: CircluarLinkedList/Node.cs (hunk 2 above)
Comment (BDullanty) [reply CircluarLinkedList/Node.cs:38]: When checking out why we did not implement it like you mention, I get this error:

image

Which explains why we have our implementation instead

File: CircluarLinkedList/CircluarLinkedList.csproj
Comment (quattro004) [inline CircluarLinkedList/CircluarLinkedList.csproj:1]: The instructions say: Create a class library project called "GenericsHomework.". I noticed the solution file has that name but I think that this project and its tests should as well.

**Review by quattro004 — APPROVED** [review]
Nice work, I like the how the usage of Theory in the tests kept the code clean.

**Pull request closed without merging**